elastic/kibana
# [Fleet] Fix ES index patterns for custom package

## Summary

Resolve #174455

Add the es_index_patterns when adding installing input package assets for a dataset so it will be display in the log explorer app `/app/observability-logs-explorer`

While looking at that code I use the opportunity to remove and unused exposed service `esIndexPattern` 

I add to do some refacto (moving function to their own module) to make the `installAssetsForInputPackagePolicy` testable.

## How to tests

Try to add a custom log integration with a generic dataset or one you create and check it's added to the log explorer

<img width="700" alt="Screenshot 2024-01-31 at 1 52 27 PM" src="https://github.com/elastic/kibana/assets/1336873/1fb19509-4e1b-4e3a-a7d7-aeabe6f9d30d">
<img width="700" alt="Screenshot 2024-01-31 at 1 52 07 PM" src="https://github.com/elastic/kibana/assets/1336873/e6da5352-ae32-4384-82d0-b79afb90ffee">
